### PR TITLE
[1.5.0] Update docs for dynamic webhooks

### LINKED ISSUE #314: [Docs] Add doc for dynamic webhooks
**Is your feature request related to a problem? Please describe current state:**
 

See https://github.com/kyverno/kyverno/pull/2425.

### PR DESCRIPTION
**Pull request by realshuting**
Signed-off-by: ShutingZhao <shutting06@gmail.com>

Closes #314.
Closes https://github.com/kyverno/kyverno/issues/1983.

### CODE REVIEW

File: content/en/docs/Monitoring Kyverno With Prometheus Metrics/policy-rule-info-total.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ kyverno_policy_rule_info_total
 | policy\_namespace        |                                  | Namespace in which this Policy resides (only for policies with kind: Policy), For ClusterPolicies, this field will be "-"                         |
 | rule\_name               |                                  | Name of the rule, in the above policy, which is evaluating in this situation                                                                      |
 | rule\_type               | "validate", "mutate", "generate" | Rule's behaviour type.<br>For rule\_execution\_cause="background\_scan", it will always be "validate" as background scans only run validate rules |
+| status\_ready            | "true", "false"                  | Readiness of the policy. When ready, the policy is able to serve the admission request
```
Comment (JimBugwadia) [inline content/en/docs/Monitoring Kyverno With Prometheus Metrics/policy-rule-info-total.md:35]: Readiness of the policy. When ready, the policy is able to serve admission requests

File: content/en/docs/Writing policies/webhooks.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+title: Dynamic Webhook Configurations
+description: Manage webhook configurations dynamically.
+weight: 2
+---
+
+From Kyverno 1.5.0, the `mutatingWebhookConfiguration` and the `validatingWebhookConfiguration` are registered and managed dynamically based on the configured policies. Prior to this version, Kyverno uses a wildcard webhook that allows it to receive all resources and make policy decisions. With the auto-configuration, Kyverno only receives the selected resources to reduce unnecessary admission requests being forwarded to Kyverno. This feature is enabled by default in 1.5.0 and can be turned off by flag `--autoUpdateWebhooks=false`. If disabled, Kyverno creates the default webhook configurations that allows all resources and with `FailurePolicy` set to `Ignore`.
```
Comment (JimBugwadia) [inline content/en/docs/Writing policies/webhooks.md:7]: Starting with Kyverno 1.5.0, the `mutatingWebhookConfiguration` and the `validatingWebhookConfiguration` resources are registered and managed dynamically based on the configured policies. Prior to this version, Kyverno uses a wildcard webhook that allowed it to receive admission requests for all resources. With the webhook auto-configuration feature, Kyverno now only receives admission requests for select resources hence preventing unnecessary admission requests being forwarded to Kyverno. 

This feature is enabled by default in 1.5.0 and can be turned off by flag `--autoUpdateWebhooks=false`. If disabled, Kyverno creates the default webhook configurations that forwards admission requests for all resources and with `FailurePolicy` set to `Ignore`.

File: content/en/docs/Writing policies/webhooks.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## Configurable Fields
```
Comment (JimBugwadia) [inline content/en/docs/Writing policies/webhooks.md:9]: ## Policy configurable Fields

File: content/en/docs/Writing policies/webhooks.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+`pol.spec.failurePolicy`: FailurePolicy defines how unrecognized errors from the admission endpoint are handled. Rules within the same policy share the same failure behavior. Allowed values are Ignore or Fail. Defaults to Fail.
```
Comment (JimBugwadia) [inline content/en/docs/Writing policies/webhooks.md:11]: FailurePolicy defines how unrecognized errors from the admission endpoint are handled by the Kubernetes API server.

File: content/en/docs/Writing policies/webhooks.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+`pol.spec.webhookTimeoutSeconds`: WebhookTimeoutSeconds specifies the webhook timeout for this policy. After the timeout passes, the admission request will fail based on the failure policy. The default timeout is 10s, the value must be between 1 and 30 seconds.
```
Comment (JimBugwadia) [inline content/en/docs/Writing policies/webhooks.md:13]: After the configured time has expired, the admission request will be handled based on the failure policy. The default timeout is 10s, the value must be between 1 and 30 seconds.

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
So, with this feature will all existing installations switch to `Fail` from `Ignore`? Or, will this only be the default behavior for new installations?

**Comment by realshuting** [comment]
> So, with this feature will all existing installations switch to `Fail` from `Ignore`? Or, will this only be the default behavior for new installations?

Yes, the existing webhooks will be switched to `Fail`.

File: content/en/docs/Installation/_index.md [hunk 6]
```diff
@@ -403,7 +404,10 @@ To modify the ConfigMap, either directly edit the ConfigMap `kyverno` in the def
 
 ### Webhooks
 
-By default, the Kyverno webhook will process all API server call-backs from all Namespaces. In some cases, it is desired to limit those to certain Namespaces based upon labels. Kyverno can filter on these Namespaces using a `namespaceSelector` object by adding a new `webhooks` object to the ConfigMap. For example, in the below snippet, the `webhooks` object has been added with a `namespaceSelector` object which will filter on Namespaces with the label `environment=prod`. The `webhooks` key only accepts as its value a JSON-formatted `namespaceSelector` object.
+Starting with Kyverno 1.5.0, the webhooks are managed dynamically based on the configured policies, see [Dynamic Webhook Configurations]((/docs/writing-policies/webhooks/)) for details. Disable auto-update by setting flag `--autoUpdateWebhooks=false` to manually configure `namespaceSelector` and other fields.
```
[the remark this answers is not in the document]
Comment (realshuting) [reply content/en/docs/Installation/_index.md]: Updated, thank you @chipzoller !

**Review by JimBugwadia — APPROVED** [review]

**Pull request merged**